rfcs/0167-maintainer-account.md

```diff
@@ -0,0 +1,270 @@
+
+# Detailed design
+
+[design]: #detailed-design
+
+At the time of this writing, the primary nixpkgs hosting platform is GitHub, but this RFC is meant to be platform agnostic.
+If in the future nixpkgs moves to an alternative platform, these requirements and expectations should transfer to the new platform as well.
+
+## Update maintainers/README
+
+Update the section `How to lose maintainer status` in the [maintainers/README.md](https://github.com/NixOS/nixpkgs/tree/master/maintainers/README.md)` file:
+
+```md
+### How to become a maintainer
+
+We encourage people who care about a package to assign themselves as a maintainer.
+Commit access to the nixpkgs repository is not required to become a maintainer.
+
+Maintainers are required to have an account on the nixpkgs hosting platform, which is currently GitHub.
+You are expected to maintain an active account on the nixpkgs hosting platform, and are required to provide this information when adding yourself to `maintainer-list.nix`.
+
+In order to do become a maintainer, add yourself to the [`maintainer-list.nix`](./maintainer-list.nix), and then to the desired package's `meta.maintainers` list, and send a PR with the changes.
+
+### How to lose maintainer status
+
+The following spells out reasons why a maintainer may lose this status.
+This removal ensures an accurate view of the state of maintenance of packages, including who will update the package or respond to issues and pull requests.
+If a maintainer is removed for inactivity or unreachability, this is not permanent and can be reverted by the maintainer at any time.
+
+#### Inactivity
+
+Maintainers who have become inactive on a given package will be removed as maintainer of the package in question.
+
+If we notice that the author hasn't reacted to package-related notifications for more than 3 months, or 2 package releases, whichever is longer, then they can be removed.
+
+Removing the maintainer happens by making a pull request on the package, adding that person as a reviewer, and then waiting a minimum of one week for response or feedback.
+
+#### Unreachable
+
+Maintainers who do not maintain an active account on the nixpkgs hosting platform will be removed as maintainers.
+An effort will be made to contact them through information listed in `maintainer-list.nix` prior to removal.
+Individuals will receive a window of at least one week to respond to contact attempts.
+
+Maintainers who have provided an email address as a point of contact, which when contacted are not available through this email address (e.g. emails bounce and fail to deliver), will have this email address removed. They must be added as a reviewer on the corresponding pull request and given one week for response or feedback prior to merging.
+
+#### Violation of Code of Conduct
+
+Maintainers are not immune to the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and must be held to the same standard as non-maintainers.
```

Comment:
```suggestion
```

Don't think we need to repeat this in every document, as the COC explicitly mentions "contributors and maintainers".

Reply:
What is the cost of repeating it?

Reply:
It's redundant, do we need to link to every policy in every other policy? The longer something is, the less peopele read it.

Reply:
I don't think it's a big deal as this is at the end, if they don't read it, so be it.
This reason for not reading things does not matter, what matters is that it becomes policy and can be enforced.

Reply:
I think it’s worthwhile to repeat here. This is where we describe how a maintainer can be removed, and it seems better to explain all the reasons rather than expect people to look in multiple places.

Reply:
> This reason for not reading things does not matter, what matters is that it becomes policy and can be enforced.

Sorry if I've missed something since [the COC was introduced](https://discourse.nixos.org/t/new-code-of-conduct-discussion/35179/47), but it's hard not to read this as an another attempt to sneak a COC in through a side door. Isn't the current claim still that the COC is strictly applicable to the moderation team or has that changed?

Reply:
It is my understanding that permanent bans by the moderation team also apply to GitHub. In which case this is already being applied. If I'm wrong, I'd be happy to be corrected, but if such a ban does block them from GitHub then they will not qualify as a maintainer any longer.

Reply:
> it seems better to explain all the reasons rather than expect people to look in multiple places.

I agree; the very next sentence explains the reason, which is why I didn't suggest removing that.

> Isn't the current claim still that the COC is strictly applicable to the moderation team or has that changed?

As per how I read the linked discussion, the COC is documenting (mostly) how the moderation team currently moderates. It would therefore apply to all participants in the community.

Reply:
I don't think the current language is strictly inaccurate but it seems very easy to misinterpret. The harm of redundancy might be more obvious were it more precise:

> Maintainers are not immune to actions taken by the moderation team, such as applications of [their Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md), and must be held to the same standard as non-maintainers.

Reply:
As far as I'm aware, if the moderation team perma-bans someone they ban them on matrix, discourse and github. I don't really get the discussion about leaving the sentence in or not, the CoC applies anyways and as raito said having one sentence more in the document doesn't really cost anything.

Reply:
Technically, ban on GitHub causes inability to react, but indeed having this separate case permits applying changes based on announced long-term bans without waiting a couple of months here or a couple of weeks there. And as the item in the list is useful, one more sentence in it doesn't seem to harm much, indeed.

Reply:
If I understand correctly, the reason we want this sentence is that somebody can't say, "*You can't ban me because then I can't fulfill my duties as enumerated in RFC 167!*" Fair enough, but I think the sentence as written is both inadequate to that purpose and misleading in other directions.

It's inadequate because AFAIK the COC is not necessarily comprehensive of all moderation rationale's. Somebody could still say, "*You can't ban me because according to RFC 167 only the COC can justify moderation of maintainers and I found some way to be annoying which weasels around the COC!*" Granted, it's hard to imagine the current COC not justifying any moderator action but a future COC might not be so vague. My point is that the COC would seem to be irrelevant to the moderation team's authority to ban people.

It's misleading because it uses the term "NixOS Code of Conduct" which implies that the NixOS community has adopted a COC which it has not. And it uses the legal term "immunity" which implies that the COC is a binding document, when the COC isn't even binding on moderators.

Language which is more to the point would avoid these issues:

```suggestion
Maintainers are not exempt from moderation actions, even if said actions inhibit or prevent them from fulfilling their duties as maintainers.
```

Reply:
I disagree, I expect maintainers to adhere to the NixOS CoC. If you want to make this a blocker point, join the shepherd team.

Reply:
@ryneeverett's proposals of wording sound reasonable to me. I think that framing this in the context of moderation instead of CoC is a good idea.

Reply:
After thinking more about it, I concede this point, I think the suggesting change is good.

Reply:
The CoC wording as written would give RFC grounds to two annoying demands generally rejected by the moderation team — justify moderation actions specifically via CoC (note that currently CoC is a piece of documentation downstream from moderation policies…), and justify that «same standard» is applied. The proposed wording does fit better the setup installed by the moderation team RFC.

Reply:
> I think that framing this in the context of moderation instead of CoC is a good idea.

Just want to note that there is a following sentence, that was not touched here, that explicitly still mentions COC:

> Violations of the Code of Conduct severe enough to warrant permanent removal will incur removal as maintainer as well.

My feedback didn't apply to this sentence, but it seems like other feedback on this thread _does_.

Reply:
```suggestion
#### Community Moderation

Actions taken through community moderation can impact your ability to remain a maintainer. Maintainers who have been permanently banned from NixOS spaces through moderation will be also removed as maintainers.
```

Reply:
Done.